number_parser/parser.py

```diff
@@ -252,20 +259,37 @@ def parse_number(input_string, language=None):
     if language is None:
         language = _valid_tokens_by_language(input_string)
 
-    lang_data = LanguageData(language)
+    if numeral_systems is None:
+        numeral_systems = NUMERAL_SYSTEMS
+
+    if _search_roman(input_string):
+        numeral_systems = ['roman']
```

Comment:
`is_roman` may be better, since we are checking if the whole number is a valid Roman number, not if the string contains a Roman number. Something with a Roman number but non-Roman characters would not match here.

Also, this code ignores the input `numeral_systems` parameter. If it does not contain `roman`, this code should not do this, and instead should probably return `None` directly. This scenario is probably worth a test.

Similar (but opposite) with `decimal` below.